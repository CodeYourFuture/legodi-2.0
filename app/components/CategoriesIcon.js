import React from 'react'
import {
  Text,
  StyleSheet,
  TouchableOpacity
} from 'react-native'
import { Ionicons } from '@expo/vector-icons'

const CategoriesList = (props) => {
  const getIconName = (name) => {
    let iconName = ''
    switch (name) {
      case 'Welcome':
        iconName = 'md-information-circle'
        break
      case 'Asylum':
        iconName = 'md-paper'
        break
      case 'Volunteering':
        iconName = 'md-briefcase'
        break
      case 'Food':
        iconName = 'md-pizza'
        break
      case 'Parks':
        iconName = 'md-paw'
        break
      case 'Sport':
        iconName = 'md-tennisball'
        break
      case 'Shopping':
        iconName = 'md-cart'
        break
      case 'Transport':
        iconName = 'md-bus'
        break
      case 'Children':
        iconName = 'md-contacts'
        break
      case 'Emergency':
        iconName = 'md-warning'
        break
      default:
        iconName = 'md-help'
    }
    return iconName
  }

  return (
    <TouchableOpacity style={styles.box} onPress={() => props.navigation('Articles')}>
      <Ionicons name={getIconName(props.title)} size={36} color='#e5ba4f' />
      <Text style={styles.title}>{props.title}</Text>
    </TouchableOpacity>
  )
}

export default CategoriesList

const styles = StyleSheet.create({
  box: {
    width: 100,
    height: 100,
    marginTop: 20,
    backgroundColor: '#245e55',
    justifyContent: 'center',
    alignItems: 'center'
  },
  title: {
    color: '#fff',
    fontWeight: 'bold',
    paddingTop: 10
  }
})
