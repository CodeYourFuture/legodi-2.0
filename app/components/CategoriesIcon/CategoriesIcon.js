import React from 'react'
import {
  Text,
  TouchableOpacity,
  View
} from 'react-native'
import { Ionicons } from '@expo/vector-icons'
import styles from './IconStyles'
import PropTypes from 'prop-types'

const CategoriesIcon = (props) => {
  const getIconName = (name) => {
    let iconName = ''
    switch (name) {
      case 'Welcome':
        iconName = 'md-information-circle'
        break
      case 'Asylum':
        iconName = 'md-paper'
        break
      case 'Volunteering':
        iconName = 'md-briefcase'
        break
      case 'Food':
        iconName = 'md-pizza'
        break
      case 'Parks':
        iconName = 'md-paw'
        break
      case 'Sport':
        iconName = 'md-tennisball'
        break
      case 'Shopping':
        iconName = 'md-cart'
        break
      case 'Transport':
        iconName = 'md-bus'
        break
      case 'Children':
        iconName = 'md-contacts'
        break
      case 'Emergency':
        iconName = 'md-warning'
        break
      default:
        iconName = 'md-help'
    }
    return iconName
  }

  return (
    <TouchableOpacity
      style={styles.box}
      onPress={() => props.navigation('Articles',
        {
          id: props.id,
          categoryTitle: props.title,
          description: props.description
        }
      )}>
      <View style={styles.corner} />
      <Ionicons name={getIconName(props.iconName)} size={36} color='#e5ba4f' />
      <Text style={styles.title}>{props.title}</Text>
    </TouchableOpacity>
  )
}

CategoriesIcon.propTypes = {
  navigation: PropTypes.func,
  id: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]),
  categoryTitle: PropTypes.string,
  description: PropTypes.string,
  title: PropTypes.string
}

export default CategoriesIcon
