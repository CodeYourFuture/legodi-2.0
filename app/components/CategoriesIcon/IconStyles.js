import {
  StyleSheet
} from 'react-native'

const styles = StyleSheet.create({
  box: {
    width: 100,
    height: 100,
    marginTop: 20,
    backgroundColor: '#245e55',
    justifyContent: 'center',
    alignItems: 'center'
  },
  title: {
    color: '#fff',
    fontWeight: 'bold',
    paddingTop: 10
  }
})

export default styles
