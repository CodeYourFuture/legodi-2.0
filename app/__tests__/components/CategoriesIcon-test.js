import React from 'react'
import renderer from 'react-test-renderer'
import CategoriesIcon from '../../components/CategoriesIcon/CategoriesIcon'
import { Text } from 'react-native'
import { Ionicons } from '@expo/vector-icons'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <CategoriesIcon>
        <Ionicons name='md-information-circle' size={36} color='#e5ba4f' />

        <Text>Some Text</Text>
      </CategoriesIcon>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
