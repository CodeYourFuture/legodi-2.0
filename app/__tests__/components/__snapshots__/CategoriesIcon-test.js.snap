// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesIcon Component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#245e55",
      "height": 100,
      "justifyContent": "center",
      "marginTop": 20,
      "opacity": 1,
      "width": 100,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#fff",
        "fontWeight": "bold",
        "paddingTop": 10,
      }
    }
  />
</View>
`;

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#245e55",
      "height": 100,
      "justifyContent": "center",
      "marginTop": 20,
      "opacity": 1,
      "width": 100,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#fff",
        "fontWeight": "bold",
        "paddingTop": 10,
      }
    }
  />
</View>
`;
